src/transaction/log_recovery.c

```diff
@@ -2467,25 +2466,18 @@ log_recovery_analysis (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa, LOG_LSA * s
 
   log_page_p = (LOG_PAGE *) aligned_log_pgbuf;
 
+  is_log_page_broken = false;
   while (!LSA_ISNULL (&lsa))
     {
       /* Fetch the page where the LSA record to undo is located */
       LSA_COPY (&log_lsa, &lsa);
 
-      is_log_page_broken = false;
+      /* We may fetch only if log page not already broken, but is better in this way. */
       if (logpb_fetch_page (thread_p, &log_lsa, LOG_CS_FORCE_USE, log_page_p) != NO_ERROR)
```

Comment:
Is it possible to do the log_is_page_of_record_broken check here?

```c
      if (logpb_fetch_page (thread_p, &log_lsa, LOG_CS_FORCE_USE, log_page_p) != NO_ERROR)
	{
	  // unable to fetch the current log page.
	  is_log_page_broken = true;
	}
      if (is_media_crash == true)
	{
	  /* Check also the last log page of current record. */
	  is_log_page_broken = log_is_page_of_record_broken (thread_p, log_page_p, &log_lsa);
	}
```

Reply:
No, we need to check pages for all log records.

Reply:
I understand.